src/components/organisms/SettingsDropDown.tsx

```diff
@@ -0,0 +1,242 @@
+});
+const Flex = styled("div", { display: "flex" });
+
+const Text = styled("div", {
+  marginBottom: 20,
+  color: mauve.mauve11,
+  fontSize: 15,
+  lineHeight: 1.5,
+});
+
+const Annotation = styled("p", {
+  marginTop: 5,
+  color: mauve.mauve11,
+  fontSize: 12,
+  lineHeight: 1.5,
+});
+
+const Button = styled("button", {
+  all: "unset",
+  display: "inline-flex",
+  alignItems: "center",
+  justifyContent: "center",
+  borderRadius: 4,
+  padding: "0 15px",
+  fontSize: 15,
+  lineHeight: 1,
+  fontWeight: 500,
+  height: 35,
+
+  variants: {
+    variant: {
+      indigo: {
+        backgroundColor: indigo.indigo4,
+        color: indigo.indigo11,
+        "&:hover": { backgroundColor: indigo.indigo5 },
+        "&:focus": { boxShadow: `0 0 0 2px ${indigo.indigo7}` },
+      },
+    },
+  },
+
+  defaultVariants: {
+    variant: "indigo",
+  },
+});
+const Fieldset = styled("fieldset", {
+  all: "unset",
+  marginBottom: 15,
+  width: "100%",
+  display: "flex",
+  flexDirection: "column",
+  justifyContent: "flex-start",
+});
+
+const Label = styled("label", {
+  fontSize: 13,
+  lineHeight: 1,
+  marginBottom: 10,
+  color: indigo.indigo12,
+  display: "block",
+});
+
+const Input = styled("input", {
+  all: "unset",
+  flex: "1 0 auto",
+  borderRadius: 4,
+  padding: "0 10px",
+  fontSize: 15,
+  lineHeight: 1,
+  color: indigo.indigo11,
+  boxShadow: `0 0 0 1px ${indigo.indigo7}`,
+  height: 35,
+  "&:focus": { boxShadow: `0 0 0 2px ${indigo.indigo8}` },
+  "&::placeholder": { color: mauve.mauve7 },
+});
+
+const FieldCheck = styled("fieldset", {
+  all: "unset",
+  width: "100%",
+  display: "flex",
+  alignItems: "center",
+});
+
+const CheckLink = styled("a", {
+  color: indigo.indigo11,
+  textDecoration: "underline",
+  "&:visited": { color: indigo.indigo11 },
+});
+
+type Props = {
+  setSettingsView: React.Dispatch<React.SetStateAction<string>>;
+  settingsView: React.Dispatch<React.SetStateAction<string>>;
+  setSelectedDeviceName: React.Dispatch<React.SetStateAction<string>>;
+  selectedDeviceName: React.Dispatch<React.SetStateAction<string>>;
+  selectedDeviceIndex: React.Dispatch<React.SetStateAction<number>>;
+  setSelectedDeviceIndex: React.Dispatch<React.SetStateAction<number>>;
+  setDeviceData: React.Dispatch<React.SetStateAction<[]>>;
+  deviceData: React.Dispatch<React.SetStateAction<[]>>;
+  isLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export default function DropDownSettings({
+  setSettingsView,
+  settingsView,
+  setSelectedDeviceName,
+  selectedDeviceName,
+  selectedDeviceIndex,
+  setSelectedDeviceIndex,
+  deviceData,
+  setDeviceData,
+  isLoading,
+  setIsLoading,
+}: Props) {
+  const getDeviceName = () => {
+    let dropDownText = document.getElementById("drop_menu_settings");
+    let selectedText = dropDownText.options[dropDownText.selectedIndex].text;
+    return setSelectedDeviceName(selectedText);
```

Comment:
Please add if condition, because selectedText would be error when undefined.
```
if (selectedText) {
      return setSelectedDeviceName(selectedText);
    }
```

Reply:
Done